src/ui/components/Timeline/index.js

```diff
@@ -317,7 +317,7 @@ export class Timeline extends Component {
         // this.props.hoverTime may have changed, but it may also still result in the same paintHash
         const { paintHash: currentPaintHash } = getMostRecentPaintPoint(this.props.hoverTime);
         if (currentPaintHash === paintHash) {
-          updateTooltip({ screen, left: this.getPixelOffset(hoverTime) });
+          updateTooltip({ screen, left: this.getPixelOffset(this.props.hoverTime) });
```

Comment:
Yes, it's coming from props, but it was read before `setTimelineState` was called.
This is something I really dislike about redux: the fact that `this.props.hoverTime` and `this.props.setTimelineState({ hoverTime: value })` is like a getter/setter pair is really well hidden by redux, which leads to confusion and bugs.

Reply:
Good point, this might be a good time to move the logic to redux where we can re-fetch the state.
https://github.com/jasonLaster/webreplay-devtools/pull/new/refactor-update-timeline

anyways, it is nice to move this complex logic out of react.